test/test_helper.rb

```diff
@@ -387,4 +381,14 @@ def login_admin
       click_button "Login"
     end
   end
+
+  class ActiveRecord::Base
+    mattr_accessor :shared_connection
+    @@shared_connection = nil
+
+    def self.connection
+      @@shared_connection || ConnectionPool::Wrapper.new(:size => 1) { retrieve_connection }
+    end
+  end
+  ActiveRecord::Base.shared_connection = ActiveRecord::Base.connection
```

Comment:
@domcleal I think doing this globally might have been the cause for the long-standing jenkins tasks. I could reproduce it on katello-deploy easily. What I'm doing there now (seems to work, I've only ran the whole suite once) is to call 

```ruby
class ActionDispatch::IntegrationTest
#...
setup do
  ActiveRecord::Base.share_connection = ActiveRecord::Base.connection
end
#...
end
```

Sounds better? If it pases 2 or 3 times I guess it's safe to push and try in jenkins.